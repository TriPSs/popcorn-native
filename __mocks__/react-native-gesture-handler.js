import React from 'react'
import { View } from 'react-native'

export const TouchableNativeFeedback = class extends React.Component {

  static Ripple = jest.fn((color, borderless) => ({
    type: 'Ripple',
    color,
    borderless
  }))

  render() {
    return <View />
  }
}

export default {
  attachGestureHandler: jest.fn(),
  createGestureHandler: jest.fn(),
  dropGestureHandler: jest.fn(),
  updateGestureHandler: jest.fn(),
  State: {},
  Directions: {},
}
