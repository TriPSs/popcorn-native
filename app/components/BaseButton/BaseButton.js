import React from 'react'
import PropTypes from 'prop-types'
import { TouchableNativeFeedback } from 'react-native'

export const BaseButton = ({ children, component: Component, innerRef, rippleColor, ...rest }) => (
  <Component
    useForeground
    ref={innerRef}
    // eslint-disable-next-line babel/new-cap
    background={TouchableNativeFeedback.Ripple(rippleColor, true)}
    {...rest}>
    {children}
  </Component>
)

BaseButton.propTypes = {
  children   : PropTypes.node.isRequired,
  rippleColor: PropTypes.string,
}

BaseButton.defaultProps = {
  rippleColor: 'rgba(0, 0, 0, .3)',
  component: TouchableNativeFeedback,
  innerRef: null,
}

export default BaseButton
