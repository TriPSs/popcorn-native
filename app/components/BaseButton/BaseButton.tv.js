import React from 'react'
import { TouchableNativeFeedback } from 'react-native'

import BaseButton from './BaseButton'

export const BaseButtonTV = (...props) => (
  <BaseButton {...props} />
)

BaseButtonTV.defaultProps = {
  ...BaseButton.defaultProps,
  component: TouchableNativeFeedback,
  rippleColor: 'test',
}

export default BaseButton
