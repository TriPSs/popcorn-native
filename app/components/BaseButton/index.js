import useCorrect from 'modules/useCorrect'

import BaseButton from './BaseButton'
import BaseButtonTV from './BaseButton.tv'

export default useCorrect(BaseButton, null, BaseButtonTV)
