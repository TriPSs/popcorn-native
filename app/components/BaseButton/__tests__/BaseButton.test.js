import React from 'react'
import { shallow } from 'enzyme'
import { TouchableNativeFeedback } from 'react-native-gesture-handler'

import BaseButton from '../BaseButton'

describe('(Component) BaseButton', () => {

  it('should render with defaults', () => {
    const result = shallow(
      <BaseButton>
        test
      </BaseButton>,
    )

    expect(result).toMatchSnapshot()
    expect(TouchableNativeFeedback.Ripple).toHaveBeenCalledTimes(1)
  })

  it('should render with no ripple', () => {
    TouchableNativeFeedback.Ripple.mockClear()

    const result = shallow(
      <BaseButton
        rippleColor={null}>
        test
      </BaseButton>,
    )

    expect(result).toMatchSnapshot()
    expect(TouchableNativeFeedback.Ripple).toHaveBeenCalledTimes(0)
  })

})
