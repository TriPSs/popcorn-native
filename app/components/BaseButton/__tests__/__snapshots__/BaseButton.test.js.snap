// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) BaseButton should render with defaults 1`] = `
<TouchableNativeFeedback
  background={
    Object {
      "borderless": true,
      "color": "rgba(0, 0, 0, .3)",
      "type": "Ripple",
    }
  }
  useForeground={true}
>
  test
</TouchableNativeFeedback>
`;

exports[`(Component) BaseButton should render with no ripple 1`] = `
<TouchableNativeFeedback
  background={null}
  useForeground={true}
>
  test
</TouchableNativeFeedback>
`;
